tracing-effectful/src/Effectful/Tracing.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Effectful.Tracing
+    (
+        -- * Effect
+        Trace
+    
+        -- * Handler
+    ,   runTracing
+    ,   runTracing'
+
+        -- * Zipkin utilities
+    ,   runZipkinTracing
+    ,   withZipkin
+
+        -- * Reexport
+    ,  module Monitor.Tracing
+    ) where
+
+import Effectful
+import Effectful.Dispatch.Static
+import Control.Monad.Trace
+import Control.Monad.Trace.Class
+import Control.Monad.Catch (finally)
+import Monitor.Tracing
+import Monitor.Tracing.Zipkin (Zipkin(zipkinTracer), Settings, new, publish)
+
+-- | Provides the ability to send traces to a backend
+data Trace :: Effect
+
+type instance DispatchOf Trace = Static WithSideEffects
+newtype instance StaticRep Trace = Trace (Maybe Scope)
+
+-- | Run the 'Tracing' effect.
+--
+-- /Note:/ this is the @effectful@ version of 'runTracingT'.
+runTracing
+  :: IOE :> es
+  => Eff (Trace : es) a
+  -> Tracer
+  -> Eff es a
+runTracing actn = runTracing'  actn . pure
+
+runTracing' 
```

Comment:
```suggestion
runTracingMaybe
```

Do you remember our conversation about functions with ticks at the end? ;)

Reply:
I did, but I was just mimicking what was done in `Control.Monad.Trace` !